src/ncp/ncp_uart.cpp

```diff
@@ -287,6 +287,8 @@ void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat
     }
     va_end(args);
 
+    otNcpStreamWrite(0, reinterpret_cast<const uint8_t*>("\r\n"), 2);
```

Comment:
Each log message is currently its own emission of spinel message `VALUE_IS(STREAM_DEBUG)` from NCP. But at some point earlier, we were considering to add priority over the NCP spinel messages and possibly let longer log messages to be chopped and sent in chunks. It's not the case in the current code.

That's why wpantund treats the `STREAM_DEBUG` as a stream, and it buffers the received chunks till it sees `\n` and then dumps the log message (or it dumps the log message when it run out of buffer).   

The configuration option of `OPENTHREAD_CONFIG_LOG_SUFFIX` was added to give us flexibility to handle all these cases. 

For NCP builds intended for use with wpantund, we use a project specific configuration header file where we define `OPENTHREAD_CONFIG_LOG_SUFFIX` as `"\n"`.  

For build where posix logging is used we keep the default settings.